keel-ec2-plugin/src/test/kotlin/com/netflix/spinnaker/keel/ec2/resolvers/ImageResolverTests.kt

```diff
@@ -147,34 +149,18 @@ internal class ImageResolverTests : JUnit5Minutests {
       )
     )
 
-    fun resolve(): ResolvedImages = runBlocking {
-      subject.resolveImageId(resource)
+    fun resolve(): Resource<ClusterSpec> = runBlocking {
+      subject.invoke(resource)
     }
   }
 
   fun tests() = rootContext<Fixture<*>> {
-    derivedContext<Fixture<IdImageProvider>>("a simple image id") {
-      fixture {
-        Fixture(
-          IdImageProvider("ami-2")
-        )
-      }
+    context("no image provider") {
+        fixture { Fixture(null) }
 
-      before {
-        coEvery {
-          cloudDriverService.namedImages(any(), any(), any(), any())
-        } answers {
-          val amiId = secondArg<String>()
-          images.filter { it.tagsByImageId.containsKey(amiId) }
-        }
-      }
-
-      test("resolves to the image id") {
-        val resolved = resolve()
-        expect {
-          that(resolved.appVersion).isEqualTo("fnord-$version2")
-          that(resolved.imagesByRegion).isEqualTo(mapOf(imageRegion to imageProvider.imageId))
-        }
+      test("returns the original spec unchanged") {
+        expectThat(resolve())
+          .propertiesAreEqualTo(resource)
       }
     }
```

Comment:
`ClusterSpec.imageProvider` being `null` is now a thing. `ImageResolver` should just return the unmodified spec in that case.